### PR TITLE
[virtualization][MU]: Set PATCH_WITH_ZYPPER and UPDATE_PACKAGE in code

### PR DESCRIPTION
**Pull request by alice-suse**
This PR mainly does below:
- use var BUILD info to set UPDATE_PACKAGE in main.pm, BUILD format eg BUILD=:33310:dtb-armv7l -- :<incident-id>:<update-package-name> (libvirt pkg needs value translation)
- PATCH_WITH_ZYPPER, set to 1 for official test, and allow customized value 0 if passing down by user 

Related ticket: A4 task openqa part, https://progress.opensuse.org/issues/160856

**Note: this PR shall be merged together with https://progress.opensuse.org/issues/160859**

Verification run: 
```
verification run on http://10.67.129.77/

Unit test:
PASS: qemu pkg
qa2-dhcp-77:/var/lib/openqa/tests/sle # openqa-cli api  -X POST isos ARCH="x86_64" DISTRI="sle" VERSION="15-SP5" FLAVOR=Server-DVD-Virt-KVM-GuestMigration-Incidents TERADATA=  PATCH_WITH_ZYPPER=0  GUEST_ADMINISTRATION=0 INTERVAL_LOG=0 TEST=qam-kvm-migration-src,qam-kvm-migration-dst BUILD=":123456:qemu"
{"count":2,"failed":[],"ids":[769,770],"scheduled_product_id":315}

PASS: xen pkg
qa2-dhcp-77:/var/lib/openqa/tests/sle # openqa-cli api  -X POST isos ARCH="x86_64" DISTRI="sle" VERSION="15-SP5" FLAVOR=Server-DVD-Virt-XEN-GuestMigration-Incidents TERADATA=  PATCH_WITH_ZYPPER=0  GUEST_ADMINISTRATION=0 INTERVAL_LOG=0 TEST=qam-xen-migration-src,qam-xen-migration-dst BUILD=":123456:xen"
{"count":2,"failed":[],"ids":[777,778],"scheduled_product_id":316}

PASS: libvirt pkg
qa2-dhcp-77:/var/lib/openqa/tests/sle # openqa-cli api  -X POST isos ARCH="x86_64" DISTRI="sle" VERSION="15-SP5" FLAVOR=Server-DVD-Virt-KVM-GuestMigration-Incidents TERADATA=  PATCH_WITH_ZYPPER=0  GUEST_ADMINISTRATION=0 INTERVAL_LOG=0 TEST=qam-kvm-migration-src,qam-kvm-migration-dst BUILD=":123456:libvirt"
{"count":2,"failed":[],"ids":[779,780],"scheduled_product_id":321}

PASS: kernel-default pkg, trigger value `"BUILD" : ":123456:dtb-armv7l"`
http://10.67.129.77/tests/749/file/vars.json


PATCH_WITH_ZYPPER
- default 1: good
qa2-dhcp-77:/var/lib/openqa/tests/sle # openqa-cli api  -X POST isos ARCH="x86_64" DISTRI="sle" VERSION="12-SP5" FLAVOR=Server-DVD-Virt-XEN-GuestMigration-Incidents TERADATA= GUEST_ADMINISTRATION=0 INTERVAL_LOG=0 TEST=qam-xen-migration-src,qam-xen-migration-dst BUILD=":123456:xen"
{"count":2,"failed":[],"ids":[763,764],"scheduled_product_id":312}

- customized 0: good
qa2-dhcp-77:/var/lib/openqa/tests/sle # openqa-cli api  -X POST isos ARCH="x86_64" DISTRI="sle" VERSION="12-SP5" FLAVOR=Server-DVD-Virt-XEN-GuestMigration-Incidents TERADATA=  PATCH_WITH_ZYPPER=0  GUEST_ADMINISTRATION=0 INTERVAL_LOG=0 TEST=qam-xen-migration-src,qam-xen-migration-dst BUILD=":123456:xen"
{"count":2,"failed":[],"ids":[765,766],"scheduled_product_id":313}



Verification runs on OSD for vmware&hyperv :
- 15sp3 hyperv kernel pkg: https://openqa.suse.de/tests/15304805/file/vars.json (commented off the check of the updated pkg installed from TEST_X repo)
- 15sp4 vmware TERADATA, open-vm-tools pkg: https://openqa.suse.de/tests/15304809/file/vars.json (commented off the check of the updated pkg installed from TEST_X repo)


A3,A4 together, full verification jobs:

PASS
qa2-dhcp-77:/var/lib/openqa/tests/sle # openqa-cli api  -X POST isos ARCH="x86_64" DISTRI="sle" VERSION="15-SP5" FLAVOR=Server-DVD-Virt-KVM-GuestMigration-Incidents TERADATA=  PATCH_WITH_ZYPPER=0  GUEST_ADMINISTRATION=0 INTERVAL_LOG=0 TEST=qam-kvm-migration-src,qam-kvm-migration-dst BUILD=":123456:qemu"
{"count":2,"failed":[],"ids":[769,770],"scheduled_product_id":315}

PASS
qa2-dhcp-77:/var/lib/openqa/tests/sle # openqa-cli api  -X POST isos ARCH="x86_64" DISTRI="sle" VERSION="15-SP5" FLAVOR=Server-DVD-Virt-XEN-GuestMigration-Incidents TERADATA=  PATCH_WITH_ZYPPER=0  GUEST_ADMINISTRATION=0 INTERVAL_LOG=0 TEST=qam-xen-migration-src,qam-xen-migration-dst BUILD=":123456:xen"
{"count":2,"failed":[],"ids":[777,778],"scheduled_product_id":316}

PASS
qa2-dhcp-77:/var/lib/openqa/tests/sle # openqa-cli api  -X POST isos ARCH="x86_64" DISTRI="sle" VERSION="15-SP5" FLAVOR=Server-DVD-Virt-KVM-GuestMigration-Incidents TERADATA=  PATCH_WITH_ZYPPER=0  GUEST_ADMINISTRATION=0 INTERVAL_LOG=0 TEST=qam-kvm-migration-src,qam-kvm-migration-dst BUILD=":123456:libvirt"
{"count":2,"failed":[],"ids":[779,780],"scheduled_product_id":321}

```

### CODE REVIEW

**Comment by alice-suse** [comment]
@RoyCai7 @tbaev @martinsmarcelo @waynechen55 @Julie-CAO Hello guys, welcome to review!

File: lib/main_common.pm [hunk 1]
```diff
@@ -2434,9 +2434,37 @@ sub load_host_installation_modules {
...
+    } else {
+        $_update_package = 'kernel-default';
+    }
+
+    set_var('UPDATE_PACKAGE', $_update_package);
+    bmwqemu::save_vars();
+    diag("BUILD is $BUILD, UPDATE_PACKAGE is set to " . get_var('UPDATE_PACKAGE', ''));
+
+    # Set PATCH_WITH_ZYPPER
+    set_var('PATCH_WITH_ZYPPER', 1) unless (check_var('PATCH_WITH_ZYPPER', 0));
```
Comment (RoyCai7) [inline lib/main_common.pm:2459]: If have it, can not debug it by setting PATCH_WITH_ZYPPER=0 in the openqa-cli command. I think.

Comment (alice-suse) [reply lib/main_common.pm:2459]: Actually you still can :) . Once you set it to be 0, no matter in testsuite or trigger cmd, it won't be overwritten to 1. It is also proved in verification jobs.

Comment (RoyCai7) [reply lib/main_common.pm:2459]: yes , I see.

**Comment by RoyCai7** [comment]
LGTM

File: lib/main_common.pm [hunk 2]
```diff
@@ -2434,9 +2434,37 @@ sub load_host_installation_modules {
     loadtest "installation/reboot_after_installation";
 }
 
+sub set_mu_virt_vars {
+    # Set UPDATE_PACKAGE based on BUILD(format example, BUILD=:33310:dtb-armv7l)
+    my $BUILD = get_var('BUILD', '');
```
Comment (waynechen55) [inline lib/main_common.pm:2439]: get_required_var ?

Comment (alice-suse) [reply lib/main_common.pm:2439]: Thanks for the review. Yes, agree with you. Given that A5 task is coded on top of this commit, will update this later in another PR.

**Comment by RoyCai7** [comment]
LGTM for hyperv and vmware change.

**Comment by alice-suse** [comment]
The[ schedule_issue.py PR](https://gitlab.suse.de/qav-maintenance/auto-schedule/-/merge_requests/25) is also ready, let's merge them together.

**Pull request merged**